win32ss/user/ntuser/message.c

```diff
@@ -12,9 +12,33 @@
 DBG_DEFAULT_CHANNEL(UserMsg);
 
 #define PM_BADMSGFLAGS ~((QS_RAWINPUT << 16)|PM_QS_SENDMESSAGE|PM_QS_PAINT|PM_QS_POSTMESSAGE|PM_QS_INPUT|PM_NOYIELD|PM_REMOVE)
+#define EXCEPTION_ACCESS_VIOLATION ((DWORD)0xC0000005)
+
+/* Strings that are OK to pass between user and kernel mode
+ * There may be other strings needed that can easily be added here. */
+WCHAR StrUserKernel[3][20] = {{L"intl"}, {L"Environment"}, {L"Policy"}};
 
 /* FUNCTIONS *****************************************************************/
 
+/* PosInArray checks for strings that can pass between user and kernel mode.
+ * See: https://learn.microsoft.com/en-us/windows/win32/winmsg/wm-settingchange
+ * It mentions 'Environment', 'intl', and 'Policy'.
+ * These strings are enumerated in the StrUserKernel definition.
+ * Returns: A positive integer indicating its position in the array if the
+ * string is found, or returns a minus one (-1) if the string is not found. */
+INT PosInArray(WCHAR String[])
+{
+    INT i;
+    INT End = ARRAYSIZE(StrUserKernel);
+
+    for (i = 0; i < End; ++i)
```

Comment:
Maybe it makes sense to use a postfix increment instead, if you want to check the 1st (`[0]`) string too? As follows:
```suggestion
    for (i = 0; i < End; i++)
```
In such a case, `i` is incremented in the next iteration, so 1st row is checked too. Otherwise, the 1st checked row is `[1]`.

Reply:
Loop line is OK. The i index is incremented once the loop body has been executed, and just before redoing the `i < End` comparison.

Reply:
OK to me then. :+1: